[CI] Add inference run for the performance dashboard

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98174
* #98173

Summary: Remove fp32 training performance run and trade for amp inference
performance run.

desertfire committed:
[CI] Add inference run for the performance dashboard

Summary: Remove fp32 training performance run and trade for amp inference
performance run.

[ghstack-poisoned]

huydhn commented on .ci/pytorch/test.sh:
```diff
@@ -383,7 +385,7 @@ test_dynamo_benchmark() {
     test_single_dynamo_benchmark "training" "$suite" "$shard_id" --training --amp "$@"
   elif [[ "${TEST_CONFIG}" == *perf* ]]; then
     # Performance test training only
```
Nit: just remove this comment as this is not the case anymore

desertfire committed:
Update on "[CI] Add inference run for the performance dashboard"


Summary: Remove fp32 training performance run and trade for amp inference
performance run.

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "dashboard run has passed. CI failures are irrelevant."
